src/Common/src/System/Linq/LowLevelEnumerable.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+namespace System.Linq
+{
+    internal static class LowLevelEnumerable
+    {
+        public static bool Any<T>(this IEnumerable<T> values)
+        {
+            Debug.Assert(values != null);
+
+            IEnumerator<T> enumerator = values.GetEnumerator();
+            return enumerator.MoveNext();
+        }
+
+        public static bool Any<T>(this IEnumerable<T> values, Func<T, bool> predicate)
+        {
+            Debug.Assert(values != null);
+            Debug.Assert(predicate != null);
+            foreach (T value in values)
+            {
+                if (predicate(value))
+                    return true;
+            }
+            return false;
+        }
+
+        public static IEnumerable<U> Select<T, U>(this IEnumerable<T> values, Func<T, U> func)
+        {
+            Debug.Assert(values != null);
+
+            foreach (T value in values)
+            {
+                yield return func(value);
+            }
+        }
+
+        public static T[] ToArray<T>(this IEnumerable<T> values)
```

Comment:
`EnumerableHelpers.ToArray()` is more efficient and could be used instead.

https://github.com/dotnet/corefx/blob/master/src/Common/src/System/Collections/Generic/EnumerableHelpers.cs